src/pbrt/util/float.h

```diff
@@ -199,9 +205,9 @@ inline constexpr Float gamma(int n) {
 inline PBRT_CPU_GPU Float AddRoundUp(Float a, Float b) {
 #ifdef PBRT_IS_GPU_CODE
 #ifdef PBRT_FLOAT_AS_DOUBLE
-    return __dadd_ru(a, b);
+    return __dadd_rn(a, b);
```

Comment:
Switching to round-nearest versus round-up breaks these functions, which are used to be able to compute conservative floating-point intervals. If this is because AMD GPUs don't support this rounding mode, then the CPU path below should be used there (which doesn't give as tight bounds but is still conservative.)

(This applies to the corresponding changes in following functions as well.)

Reply:
I'll check with others whether there are some equivalents in HIP for these functions, but the ones they are in PBRT now are not supported. If there are not such functions, we will use the ones in the CPU path.